File: src/basic_training/Day15_3.java
package basic_training;

public class Day15_3 {
	public int solution(int[] num_list) {
		int count = 0;
		for(int num : num_list) {
			while(num != 1) {
				if(num % 2 == 0) num /= 2;
				else num = (num -1) / 2 ; 
				count++;
			}
		}
		return count;
	}
}

// while(조건식) 반복문
// 조건식이 true 일때만 반복문 실행

// for() 반복문
// for(int i=0; i<num_list.length; i++) {
// 	while(num_list[i] != 1) { // 원소의 값이 1 이 아닌경우 반복문 실행 (1인 경우 실행x)
// 		if(num_list[i] % 2 == 0) num_list[i] /= 2;
// 		else if(num_list[i] % 2 == 1) num_list[i] = (num_list[i] - 1) / 2;
// 		count++;
// 	}
// }

File: src/basic_training/Day15_3.js
function solution(num_list) {
	const count = 0;
	num_list.forEach((value) => {
		while (value !== 1) {
			if (value % 2 === 0) value /= 2;
			else value = (value - 1) / 2;
			count++;
		}
	})
	return count;
}


// 번외 문제풀이법 _ 배열.map() - 배열의 모든 요소에대해 연산 및 새로운 배열로 반환
// 배열 반환이 아닌 단순히 배열의 모든 요소에 대해 연산을 수행할때는 forEach 사용권장
// num_list.map((value) => {
// 	while(value !== 1) {
// 		if(value % 2 === 0) value /= 2;
// 		else value = (value - 1) / 2;
// 		count++;
// 	}
// })
